pkg/apis/resource/types.go

```diff
@@ -219,6 +221,40 @@ type BasicDevice struct {
 	//
 	// +optional
 	Capacity map[QualifiedName]DeviceCapacity
+
+	// UsageRestrictedToNode indicates if the usage of an allocation involving this device
+	// has to be limited to exactly the node that was chosen when allocating the claim.
+	//
+	// +optional
+	UsageRestrictedToNode *bool
+
+	// BindingConditions defines the conditions for proceeding with binding.
+	// All of these conditions must be set in the per-device status
+	// conditions with a value of True to proceed with binding the pod to the node
+	// while scheduling the pod.
+	// The maximum number of binding conditions is 4.
```

Comment:
Which strings are valid? This needs to be defined here and checked in validation.go.

You can say "must be a valid condition type string" and leave it to the user to figure out or link to https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Condition - but I am not sure whether a Go doc link in OpenAPI is really appropriate. Better leave it out.

 Then reuse the validation used for that type.